controllers/dogu_manager.go

```diff
@@ -54,25 +64,60 @@ type DoguRegistrator interface {
 	UnregisterDogu(dogu string) error
 }
 
+// DependencyValidator is used to check if dogu dependencies are installed
+type DependencyValidator interface {
+	ValidateDependencies(dogu *core.Dogu) error
+}
+
+// ServiceAccountCreator is used to create service accounts for a given dogu
+type ServiceAccountCreator interface {
+	CreateServiceAccounts(ctx context.Context, doguResource *k8sv1.Dogu, dogu *core.Dogu) error
+}
+
 // NewDoguManager creates a new instance of DoguManager
-func NewDoguManager(client client.Client, scheme *runtime.Scheme, resourceGenerator DoguResourceGenerator,
-	doguRegistry DoguRegistry, imageRegistry ImageRegistry, doguRegistrator DoguRegistrator) *DoguManager {
-	return &DoguManager{
-		Client:            client,
-		Scheme:            scheme,
-		resourceGenerator: resourceGenerator,
-		doguRegistry:      doguRegistry,
-		imageRegistry:     imageRegistry,
-		doguRegistrator:   doguRegistrator,
+func NewDoguManager(version *core.Version, client client.Client, operatorConfig *config.OperatorConfig) (*DoguManager, error) {
+	doguRegistry := registry.NewHTTPDoguRegistry(operatorConfig.DoguRegistry.Username, operatorConfig.DoguRegistry.Password, operatorConfig.DoguRegistry.Endpoint)
+	imageRegistry := registry.NewCraneContainerImageRegistry(operatorConfig.DockerRegistry.Username, operatorConfig.DockerRegistry.Password)
+	resourceGenerator := resource.NewResourceGenerator(client.Scheme())
+
+	cesRegistry, err := cesregistry.New(core.Registry{
+		Type:      "etcd",
+		Endpoints: []string{fmt.Sprintf("http://etcd.%s.svc.cluster.local:4001", operatorConfig.Namespace)},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create ces registry: %w", err)
+	}
+
+	doguRegistrator := NewCESDoguRegistrator(client, cesRegistry, resourceGenerator)
+	dependencyValidator := dependency.NewCompositeDependencyValidator(version, cesRegistry.DoguRegistry())
+
+	clientSet, err := kubernetes.NewForConfig(ctrl.GetConfigOrDie())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create clientSet: %w", err)
 	}
+
+	executor := resource.NewCommandExecutor(clientSet, clientSet.CoreV1().RESTClient())
+	serviceAccountCreator := serviceaccount.NewServiceAccountCreator(cesRegistry, executor)
```

Comment:
Now it is used only in the creator. In the future it will be although used in a component responsible for deleting service accounts or calling upgrade notifications.